cpp/include/cudf/ast/detail/operators.hpp

```diff
@@ -35,14 +35,14 @@ namespace ast {
 
 namespace detail {
 
-// Type trait for wrapping nullable types in a thrust::optional. Non-nullable
+// Type trait for wrapping nullable types in a cuda::std::optional. Non-nullable
 // types are returned as is.
 template <typename T, bool has_nulls>
 struct possibly_null_value;
 
 template <typename T>
 struct possibly_null_value<T, true> {
-  using type = thrust::optional<T>;
+  using type = cuda::std::optional<T>;
```

Comment:
Is there a reason why we didn't do this before in #15091 or #16604?

Reply:
Here, I might have answered it myself. https://github.com/rapidsai/cudf/pull/15091#issuecomment-2004286213

> Device uses of `thrust::optional` can be replaced with `cuda::std::optional` in a future PR, once we upgrade to CCCL 2.3 or newer

Is that why?

Reply:
Nope. We saw test failures with `cuda::std::optional` here. https://github.com/rapidsai/cudf/pull/15091#issuecomment-2297148545